dbeam-core/src/main/java/com/spotify/dbeam/options/JdbcExportArgsFactory.java

```diff
@@ -117,9 +117,11 @@ public static QueryBuilderArgs createQueryArgs(JdbcExportPipelineOptions options
 
   private static QueryBuilderArgs createQueryBuilderArgs(JdbcExportPipelineOptions options)
       throws IOException {
+    checkArgument((options.getTable() != null) != (options.getSqlFile() != null),
+            "Either --table or --sqlFile must be present");
     if (options.getSqlFile() != null) {
       return QueryBuilderArgs.create(
-          options.getTable(), PasswordReader.readFromFile(options.getSqlFile()));
+          "user_based_query", PasswordReader.readFromFile(options.getSqlFile()));
```

Comment:
Is there a ways users can configure `"user_based_query"` to something else?

Another thought, maybe `table` name is not important. We could consider having two different implementations of `QueryBuilderArgs`: one "table name" based and another `sqlFile` based. Not that we necessary need to solve it in this PR, just sharing a thought and asking for your thoughts.

Reply:
That was the original idea! however I  found table name used in several places for several purposes (part of JobConfiguration, QueryBuilder and  schema generation). I think it would be useful to create more explicit parameters for this purposes. We may want to add options for each and used tablename as default. can you create a story for it ? I can help with it

Reply:
I would go even further and remove:

- QueryBuilderArgs.tableName()
- QueryBuilderArgs.Builder.setTableName()

and remove _tableName_ parameter from:

- QueryBuilderArgs.create(String tableName, String sqlQueryOpt)

Reply:
I see few options to get rid off tableName.

- keep  setTableName and remove getTableName(), at the time QueryBuilderArgs creation if tablename is present, baseSqlQuery can be set based on tablename. Then queryBuilder would work only on top of baseSqlQuery rather than 2 different ways. Cons: passing additional logic to QueryBuilderArgs creation to build  base sql
- remove setTableName and getTableName from QueryBuilderArgs. Then we will need to pass additional table param to QueryBuilder from DBeamPipelineOptions. Cons: I think QueryBuilderArgs lose its purpose if we need inject additional params to QueryBuilder outside of this
- Remove support for tablename and force sqlFile usage. I would not go for this option, I think it is a nice feature to have this. 

Let me know what you think

Reply:
1. Exporting `--table` parameter is an important use case. Personally I think it is more important than `--sqlFile`. Regardless, I think there is space for both on DBeam.
2. When breaking APIs (e.g. removing a parameter), we would need to have a good justification to introduce such friction on DBeam users. I don't think it is the case here.
3. I think `QueryBuilderArgs` needs some refactoring, it is the only (non builder) class that is mutable. I would like to see that part immutable and with a better logic for both tableName and sqlFile.

I think don't yet see a `QueryBuilderArgs` without `tableName` parameter. Maybe it is possible, maybe worth to try.

Reply:
The code LGTM, we can address this and continue discussion of improvements in a separate PR.